packages/back-end/src/controllers/datasources.ts

```diff
@@ -513,14 +513,8 @@ export async function putDataSource(
     });
     return;
   }
-  // Require higher permissions to change connection settings vs updating query settings
-  const permissionLevel = params
-    ? "createDatasources"
-    : "editDatasourceSettings";
-  req.checkPermissions(
-    permissionLevel,
-    datasource?.projects?.length ? datasource.projects : ""
-  );
+
+  req.checkPermissions("editDatasourceSettings", datasource.projects);
```

Comment:
It was - the frontend of `[did].tsx` was checking for `createDatasources` to determine whether or not to render the `Edit Buttons`.

Once that was updated to check against the `editDatasourceSettings` permission, this was causing trouble as `params` were always included in the payload, even if they didn't change, so this ternary never evaluated to `editDatasourceSettings`.

But, before the updates to `[did].tsx` were made, this issue never surfaced since we were checking for `createDatasource` permission there.